CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# Contributing to SkyPilot
+
+Thank you for your interest in contributing to SkyPilot! We welcome and value 
+all contributions to the project, including but not limited to: 
+
+* [Bug reports](https://github.com/skypilot-org/skypilot/issues) and [discussions](https://github.com/skypilot-org/skypilot/discussions)
+* [Pull requests](https://github.com/skypilot-org/skypilot/pulls) for bug fixes and new features
+* Test cases to make the codebase more robust
+* Examples
+* Documentation
+* Tutorials, blog posts and talks on SkyPilot
+
+Please read through this document before contributing so that there is all the necessary information needed for us to effectively respond to your contribution.
+
+## Contributing Code
+
+We use Github to track issues and features. For new contributors, we recommend looking at issues labeled ["good first issue"](https://github.com/sky-proj/sky/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22+).
+
+### Installing SkyPilot for development
+```bash
+# SkyPilot requires python >= 3.6 and < 3.10.
+# You can just install the dependencies for
+# certain clouds, e.g., ".[aws,azure,gcp]"
+pip install -e ".[all]"
+pip install -r requirements-dev.txt
+```
+IMPORTANT: Please `export SKYPILOT_DEV=1` before running the CLI commands in the terminal, so that developers' usage logs do not pollute the actual user logs.
+
+### Environment variables for developers
```

Comment:
Proposal: how about we move this to the end of this doc?  It's more important to read about things like Testing, style guide, etc.

Section order that feels more natural to me (from here on):
- Submitting pull requests
- Testing
- Some general engineering practice suggestions
- Environment variables for developers

Reply:
Sounds good! I do want to keep testing before submitting pull requests, so the order now is:

- Installing
- Testing
- Submitting pull requests
- Some general engineering practice suggestions
- Environment variables for developers